### ISSUE TITLE
Create HD of web images to send to Zhejiang for mirror site

### ISSUE THREAD

**Issue by TBRC-Travis** [buda-base/archive-ops#118]

**Comment by jimk-bdrc** [comment]
Hi, I'll get the cost and work with @Dechen1  on the funding.

**Comment by Dechen1** [comment]
Thanks Travis for opening an issue. I think we should choose a HD that "will be especially good at withstanding all the shocks and temperature changes involved in the journey to Hangzhou" -quote Jann

**Comment by xristy** [comment]
Here's a spreadsheet as of 20200709 of the Works in the database with their on-disk status and restrictions and so on.

[TBRC_WORKS_DETAILS-20200709.xlsx](https://github.com/buda-base/archive-ops/files/4898565/TBRC_WORKS_DETAILS-20200709.xlsx)

This should be useful in selecting what is wanted. Once a list is obtained, ther4e is presumably some AO tool for computing the expected size of the disk needed.

**Comment by jimk-bdrc** [comment]
```bash
cat *csv | cut -f1 -d, | parallel 'du -s -BM /mnt/rs5Archive/{}' | tee sizes
```

**Comment by jimk-bdrc** [comment]
@Dechen1 advises: Additional hard drive space cost: 1TB=6000CYN per year (about $857 per year), It is expensive, but we can afford another 3-5TB if needed
We have 5TB space right now.

**Comment by jimk-bdrc** [comment]
Noted that many works which have 'ondisk = true' aren't on our archive. They may be in  S3.

**Comment by TBRC-Travis** [comment]
Good point.  Most notably for NLM and FPL web images not yet transferred to RS5.

**Comment by jimk-bdrc** [comment]
And conversely, released works (e.g. W2DB25466) have disk = false, but are in S3 and on RS5.

**Comment by jimk-bdrc** [comment]
I used a variant of the above parallel command to create a list of image dir sizes, `RS3:/homes/home/jimk/prod/archive-ops-87/pass1/sizeStatusReleased` I summed that up  `cat pass1/sizeStatusReleased | awk -FM '{sum += $1} END { print( sum,  sum / 1024 / 1024 )}` to get the raw MB and TB count.

After all that: it's at most 3.39 TB

**Comment by xristy** [comment]
The `onDisk` column in the spreadsheet is determined as follows:
```
work/imagegroup/volume[1]/ig:description/@type = "ondisk"
```
which is set from the _Works synced to WebArchive_ script. Note: for each work, only the first imagegroup (called `volume` in the script) will be tested so there could be works where it is `not ondisk`, but other imagegroups are on disk. So some works may actually have some content on disk but  be reported as having none. I would not expect this to happen very often.

As I understand the flow, the script syncs images to S3 and enters the `ondisk`. I don't know the rest of the flow w.r.t. the RSs, so I don't know how or when the RSs are kept in sync w/ S3.

**Comment by xristy** [comment]
For cases such as `W2DB25466`, I think @TBRC-Travis uses a script where the `ondisk` info needs to be updated .

I will modify the script for the spreadsheet so that it checks imagegroups until it finds one that is  `ondisk` and reports _true_ otherwise _false_.

**Comment by jimk-bdrc** [comment]
Include SEA, per Élie

Élie Roux via gmail.com | Élie Roux via gmail.com | 12:51 PM (0 minutes ago) |   |  
-- | -- | -- | -- | --
Élie Roux via gmail.com
to me, Dechen, Jann, Travis | to me, Dechen, Jann, Travis
to me, Dechen, Jann, Travis



Merci, Élie:

Those  will be included by default. If you want to exclude them, no problem, but someone has to make the decision.

Jim 



On Tue, Jul 14, 2020 at 12:42 PM Jim Katz <jimk@tbrc.org> wrote:
What's the SEA?

Jim 

On Thu, Jul 9, 2020 at 12:49 PM Jann Ronis <jann@tbrc.org> wrote:
Hi guys,

Briefly, do we want to include the SEA material on this drive?

Jann

**Comment by Dechen1** [comment]
Conversation about AWS S3 and NAS from Guli:
—————  2020-7-11  —————

古迦陵 6:53 AM
Hi Elie, I have made some study on Synology NAS product, Amazon EC2 and Amazon EFS(Amazon Elastic File System).

For Synology NAS, maybe it is not so convenient to directly mount it to the VM cause the VM is only accessible via VPN from home. So I think it should be possible to mount it to my computer(mac os system) first, and then upload the file to the VM.

For AWS, I have registered an account, created a EC2 instance of Redhat system. Then I create a Amazon EFS instance, an mount it to the EC2 by NFS. Then I try to sync data from EC2 in the VM of BUDA, and it works. But I have a question that for the synchronization of the updated data of BUDA,should we transfer the whole image file each time, or just upload the incremental data is enough for the updating.

I have draw two diagrams show how it works.

古迦陵 6:53 AM
nas_aws_solution.pptx(View in attachment)

古迦陵 6:57 AM
The EC2 instance is now accessible, by using command: ssh root@3.129.8.61
The root password is: 4rfcGty@123!

古迦陵 6:58 AM
ssh port is 22

Elie Roux 10:20 AM
Oh ok that's interesting thanks! Perhaps the easiest would be that we use your ec2 instance to transfer the images from our aws s3 to yours, and then you can transfer these from your s3 to the server on ZJU... that would be associated with a cost though... can you look on the aws pricing in China how much transfering 5TB twice would be?

Elie Roux 10:21 AM
I guess we could also explore giving you a temporary access to s3 so that you can download the images on the ZJU server directly but my guess is that the firewall won't like it...

古迦陵 10:51 AM
the firewall is a problem, but we can give it a try though.

古迦陵 11:13 AM
5TB data transfer cost:

1.transfer twice:
AWS S3 US -&gt; AWS S3 Hongkang
0.09 * 5 * 1024 = 460.8 USD

AWS S3 Hongkang -&gt; ZJU BUDA
0.12 * 5 * 1024 = 614.4 USD

2.directly transfer:
AWS S3 US -&gt; ZJU BUDA
0.12 * 5 * 1024 = 614.4 USD

Elie Roux 11:36 AM
Ok thanks a lot! It seems more cost effective to transfer a physical hard drive

古迦陵 11:39 AM
ok

**Comment by Dechen1** [comment]
[nas_aws_solution.pptx](https://github.com/buda-base/archive-ops/files/4920188/nas_aws_solution.pptx)

**Comment by TBRC-Travis** [comment]
> After all that: it's at most 3.39 TB

@jimk-bdrc one thing I remembered here is that the filesystem in which you're calculating the file size matters.  for example, your size estimate of ~3.39 TB is based on the blocksize of the btrfs filesystem (I believe), but if the files are transferred to a FAT filesystem with a different blocksize, the resulting filesize usage will be different.  just something to keep in mind as this has bitten us in the past when transferring data across different filesystems.

**Comment by jimk-bdrc** [comment]
We could get around this by zipping, but that's a good caution.

**Comment by eroux** [comment]
@Dechen1 this is a public place and you've just given the password to a server Guli own and will pay the use of... that's really not good for him... can you ask him to change the password?

**Comment by Dechen1** [comment]
Oh, I see. This is something I have been concerned about. All the posts on Buda-base are public? I thought we have internal posts if someone creates an issue.

**Comment by Dechen1** [comment]
Oh, I see. This is something I have been concerned about. All the posts on
Buda-base are public? I thought we have internal posts if someone creates
an issue.



Dechen Pema | Director of Administration
ནང་བསྟན་དཔེ་ཚོགས་ལྟེ་གནས། Buddhist Digital Resource Center 佛教数字资源中心
1430 Massachusetts Ave 5th Floor Cambridge, MA 02138
Tel: 617-354-7900 | tbrc.org | donate.tbrc.org

**Comment by Dechen1** [comment]
I just delete the post.

**Comment by xristy** [comment]
Please ask Guli to change the password. We do not know who may have seen the post prior to deletion. Security first please.
